runtime/nativeMain/src/kotlinx/benchmark/NativeBlackhole.kt

```diff
@@ -1,35 +1,89 @@
 package kotlinx.benchmark
 
-import kotlinx.cinterop.pin
+import kotlinx.cinterop.toByte
+import kotlin.concurrent.Volatile
+import kotlin.native.identityHashCode
+import kotlin.random.Random
 
+@OptIn(ExperimentalStdlibApi::class)
 actual class Blackhole {
-    actual fun consume(obj: Any?) {
-        obj?.pin()
+    @Volatile
+    var i0: Int = Random.nextInt()
+    var i1 = i0 + 1
+
+    @Volatile
+    var l0 = Random.nextLong()
+    var l1 = l0 + 1L
+
+    @Volatile
+    var f0 = Random.nextFloat()
+    var f1 = f0 + 1.0f
+
+    @Volatile
+    var d0 = Random.nextDouble()
+    var d1 = d0 + 1.0
+
+    @Volatile
+    var bh: Blackhole? = null
+
+    actual inline fun consume(obj: Any?) {
+        val hc = obj?.identityHashCode() ?: 0
```

Comment:
Also, we can split null and non-null paths into two methods if it makes any sense

Reply:
Well, `identityHashCode` is defined for `Any?` and `Returns: 0 for null object, identity hash code otherwise.`, so I removed all the extra null checks.
Not sure if a separate path really needed, especially if it differs from `consume(Int)` (it may lead to different performance depending on how frequently we receive nulls).

Reply:
Even better, thanks